libraries/engage/core/helpers/i18n.js

```diff
@@ -0,0 +1,58 @@
+import {
+  getTranslator,
+  getPriceFormatter,
+  getDateFormatter,
+  getTimeFormatter,
+  getNumberFormatter,
+} from '@shopgate/pwa-common/helpers/i18n';
+import { logger } from '@shopgate/pwa-core/helpers';
+
+/**
+ * @typedef {Object} I18nHelpers
+ * @property {Function} init - Inits locales.
+ * @property {Function} text - Text translator.
+ * @property {Function} price - Price translator.
+ * @property {Function} date - Date translator.
+ * @property {Function} time - Time translator.
+ * @property {Function} number - Number translator
+ */
+/**
+ * I18n helpers.
+ * @returns {I18nHelpers}
+ */
+const I18n = () => {
+  let didInit = false;
+  /**
+   * Function that throws when called. Used in a default instance of i18n to notify developer
+   * when the module is used without initialization.
+   */
+  const notReadyCb = () => {
+    throw new Error('i18n is not yet ready');
```

Comment:
Throwing Error outside of `ErrorBoundary` would lead to frozen AppStart. Better log error instead of throwing it

Reply:
Won't do as discussed in comments below.